sonarts-core/docs/rules/no-redundant-jump.md

```diff
@@ -0,0 +1,31 @@
+# no-redundant-jump
+
+Jump statements, such as `return`, `break` and `continue` let you change the default flow of program execution,
+but jump statements that direct the control flow to the original direction are just a waste of keystrokes.
+
+## Noncompliant Code Example
+
+```typescript
+function redundantJump(x: number) {
+  if (x == 1) {
+    console.log("x == 1");
+    return;
+//  ^^^^^^^ {{Remove this redundant jump.}}
+  }
+}
+```
+## Compliant Solution
+
+```typescript
+function redundantJump(x: number) {
+  if (x == 1) {
+    console.log("x == 1");
+  }
+}
+```
+## Exceptions
+
+`break` and `return` inside `switch` statement are ignored, because they are often used for consistency.
+`continue` with label is also ignored, because label is usually used for clarity.
```

Comment:
@saberduck break with label is ignored as well, right?

Reply:
`break` in loop always changes control flow, so in fact there is never issue on `break` inside loop.  I refactored a function `isBreakOrContinueWithLabel` in the rule which was bit confusing.